engine/ingame/game_state.py

```diff
@@ -28,14 +28,22 @@ def __init__(
         game_world: world.World,
         viewport_size: Tuple[int, int],
         ingame_gui: Optional[scripts.GUI],
+        menu_gui: scripts.GUI,
+        api: scripts.GameAPI,
     ) -> None:
         self.game_world = game_world
         self.view = game_view.InGameView(game_world, viewport_size, gui=ingame_gui)
-        self.controller = game_controller.InGameController(game_world, self.view)
+        self.controller = game_controller.InGameController(
+            game_world,
+            self.view,
+            menu_gui,
+        )
+        self.api = api
 
     def setup(self) -> None:
         """Sets up the in-game state."""
         self.view.setup()
+        self.controller.setup(self.api)
```

Comment:
A simpler approach: instead of changing the protocol to add a new method, why not just pass `api` through in the controller's constructor? That's what we do in the GUI game state.

Reply:
Yeah, I can try that. The main reason I did it like this was to follow the guidelines of the ticket